building-and-packaging/material/systempaths-and-librarytools_slides.md

```diff
@@ -0,0 +1,289 @@
+---
+SlideOptions:
+  transition: slide
+  width: 1400
+  height: 900
+  margin: 0.1
+---
+
+<style>
+  .reveal strong {
+  font-weight: bold;
+      color: orange;
+  }
+  .reveal p {
+      text-align: left;
+  }
+  .reveal section h1 {
+      color: orange;
+  }
+  .reveal section h2 {
+      color: orange;
+  }
+  .reveal section img {
+    background:none;
+    border:none;
+    box-shadow:none;
+  }
+</style>
+
+
+# System Paths, Libraries, and How to Use Them
+
+---
+
+## Introduction
+
+- Common functionality should not be redone the whole time
+    - Examples: [C++ Standard Library](https://en.cppreference.com/w/cpp), [Python Standard Library](https://docs.python.org/3/library/), [PETSc](https://petsc.org/), [FEniCS](https://fenicsproject.org/)...
+- For reusing software we need to know
+    - where the software is located
+    - how to compile and link with/against
+
+---
+
+## Static and shared libraries
+
+- Static
+    - Libraries: `NAME.a`
+    - Self-contained code/libraries
+    - Larger library/executable
+- Shared
+    - Libraries: `NAME.so`
+    - Shared code must be installed and accessible
+    - Executables can share library in memory
+
+---
+
+## Filesystem Hierarchy Standard
+
+- Defines filesystem layout and location of common files on Linux
+
+<img src="https://raw.githubusercontent.com/Simulation-Software-Engineering/Lecture-Material/add-overview-systempaths-librarytools/building-and-packaging/material/figs/filesystem_hierarchy_structure/fig.png" width=100%; style="margin-left:auto; margin-right:auto; padding-top: 25px; padding-bottom: 25px">
+
+- [Official homepage of FHS](https://refspecs.linuxfoundation.org/fhs.shtml)
+- Certain structure should
+
+**TODO** Set this link to the GitHub main branch before merging.
+
+---
+
+## Some Environment Variables
+
+- `HOME` or `~`: User's home directory
+- `PWD`: Current working directory
+- `PATH`: Location of executable
+- `LC_<NAME>`: Locale's
+- `LANG`: Systems' language
```

Comment:
At the moment I would vote for keeping them in as the slide's length is fine. I put these variables in, in case I use a fresh VM to print environment variables. On the fresh VM there are few variables set, but many of these variables are `LC_*` variables. At the same time `LANG` is [strongly connected to the `LC_*` variables](https://pubs.opengroup.org/onlinepubs/7908799/xbd/envvar.html).